src/QAT/qat/purr/compiler/builders.py

```diff
@@ -65,18 +73,19 @@ def instructions(self):
         return list(self._instructions)
 
     @staticmethod
-    def deserialize(blob, model) -> "QuantumInstructionBuilder":
-        # TODO: At this point everything is a Quantum builder variation, base class losing meaning.
-        builder = QuantumInstructionBuilder(model)
-        builder._instructions = json_loads(blob, model=model)
+    def deserialize(blob) -> "QuantumInstructionBuilder":
+        builder = jsonpickle.decode(blob, context=CyclicRefUnpickler())
+        if isinstance(builder, str):
```

Comment:
this is differently handled in https://github.com/oqc-community/qat/pull/79/files#diff-57c60dad41f425dcf0b9f66368aa616df03bfab33465678710f1b2ff82b7700a